ext/php8/ddtrace.c

```diff
@@ -311,6 +311,32 @@ zval *ddtrace_spandata_property_meta(ddtrace_span_t *span) { return OBJ_PROP_NUM
 zval *ddtrace_spandata_property_metrics(ddtrace_span_t *span) { return OBJ_PROP_NUM(&span->std, 5); }
 #pragma GCC diagnostic pop
 
+BOOL_T ddtrace_fetch_prioritySampling_from_root(int *priority) {
+    zval *priority_zv;
+    ddtrace_span_fci *root_span = DDTRACE_G(open_spans_top);
+
+    if (!root_span) {
+        return false;
+    }
+
+    while (root_span->next) {
+        root_span = root_span->next;
+    }
+
+    zval *root_metrics = ddtrace_spandata_property_metrics(&root_span->span);
+    ZVAL_DEREF(root_metrics);
```

Comment:
What scenario would cause a zval from `OBJ_PROP_NUM` to be `IS_REFERENCE`?

Reply:
`$someObj->theProp = &$myVar;` - trivial actually.

Reply:
Ah, of course. 😆 `E_CANNOT_SEE_FOREST_FOR_THE_TREES`